docs/providers.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Providers
+toc: true
+weight: 400
+indent: true
+---
+
+# Providers
+
+Providers are external infrastructure providers such as Amazon Web Services,
+Microsoft Azure, Google Cloud and Alibaba Cloud.
```

Comment:
Should we mention that providers extend Crossplane? I suspect people who are not familiar with Crossplane could interpret "provider" in this opening sentence to just mean "cloud provider", without realising that we're talking about providers as a Crossplane concept.